File: chatwoot/ruby/src/ListAllConversationStatisticsExample.rb
require "json"
require "chatwoot_client"

ChatwootClient.configure do |config|
    config.api_key["userApiKey"] = "USER_API_KEY"
    # config.api_key["agentBotApiKey"] = "AGENT_BOT_API_KEY"
    # config.api_key["platformAppApiKey"] = "PLATFORM_APP_API_KEY"
end

begin
    response = ChatwootClient::ReportsApi.new.list_all_conversation_statistics(
        nil, # account_id
        nil, # metric
        nil, # type
    )

    p response
rescue ChatwootClient::ApiError => e
    puts "Exception when calling ReportsApi#list_all_conversation_statistics: #{e}"
end

File: chatwoot/typescript-axios/src/ListAllConversationStatisticsExample.ts
import axios, { AxiosError } from "axios";
import * as api from "chatwoot_client"

const configuration = new api.Configuration({
  apiKey: "USER_API_KEY",
  // apiKey: "AGENT_BOT_API_KEY",
  // apiKey: "PLATFORM_APP_API_KEY",
});

new api.ReportsApi(configuration).listAllConversationStatistics(
  0, // accountId
  "conversations_count", // metric
  "account", // type
  undefined, // id
  undefined, // since
  undefined, // until
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling ReportsApi#listAllConversationStatistics:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
